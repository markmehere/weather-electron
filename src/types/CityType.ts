export type CityType = {
  value: number; /* woeid */
  label: string; /* city, country code (e.g. Melbourne, AU) */
};