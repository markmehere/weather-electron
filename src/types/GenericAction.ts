import { Action } from "redux";

export interface GenericAction<T> extends Action {
  payload: T;
} 
