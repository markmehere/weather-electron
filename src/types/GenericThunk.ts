import { ThunkAction } from "redux-thunk";
import { AppStore } from "../redux/reducers";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type GenericThunk = ThunkAction<any, AppStore, undefined, any>;
