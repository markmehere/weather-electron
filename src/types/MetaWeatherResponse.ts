import { WeatherKind } from "./WeatherKind";

export interface ConsolidatedWeatherResponse {
  id: string;
  weather_state_name: string;
  weather_state_abbr: WeatherKind;
  created: string;
  applicable_date: string;
  min_temp: number;
  max_temp: number;
  wind_speed: number;
  wind_direction: number;
  air_pressure: number;
  humidity: number;
  visibility: number;
  predictability: number;
}

export interface MetaWeatherResponse {
  consolidated_weather: ConsolidatedWeatherResponse[]
}