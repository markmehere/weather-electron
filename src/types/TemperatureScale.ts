export enum TemperatureScale {
  Celsius,
  Fahrenheit
}
