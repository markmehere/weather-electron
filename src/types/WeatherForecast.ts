import { WeatherKind } from "./WeatherKind";

export interface WeatherForecast {
  max: number;
  min: number;
  date: string;
  kind: WeatherKind;
}
