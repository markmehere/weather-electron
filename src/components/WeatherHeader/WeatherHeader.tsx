import React from "react";
import styles from "./WeatherHeader.less";
import { ScaleToggler } from "./ScaleToggler/ScaleToggler";
import { CitySelector } from "./CitySelector/CitySelector";
import { TemperatureScale } from "../../types/TemperatureScale";
import { SetUnitActionCreator, setUnits, SetCityActionCreator, setCity, GoHomeActionCreator, goHome } from "../../redux/actions/settings";
import { AppStore } from "../../redux/reducers";
import { connect } from "react-redux";
import { CityType } from "../../types/CityType";

export interface WeatherHeaderProps {

  // The temperature scale
  scale: TemperatureScale;

  // The city
  city: CityType;

  // The home city
  homeCity: CityType | null;

  // The temperature unit action creator
  setUnits: SetUnitActionCreator;

  // The set city action creator
  setCity: SetCityActionCreator;

  // The go home unit action creator
  goHome: GoHomeActionCreator;

}

export class WeatherHeaderClass extends React.Component<WeatherHeaderProps>  {

  render() {
    const { scale, city, setUnits, setCity, goHome } = this.props;

    return <div className={styles.header}>
      <CitySelector value={city} onChange={setCity} onGoHome={this.props.homeCity ? goHome : undefined} />
      <ScaleToggler value={scale} onToggle={setUnits} />
    </div>;
  }
}

const mapStateToProps = (state: AppStore) => ({
  scale: state.settings.scale,
  city: state.settings.city,
  homeCity: state.settings.homeCity
});

const mapDispatchToProps = {
  setUnits: setUnits,
  setCity: setCity,
  goHome: goHome
};

export default connect(mapStateToProps, mapDispatchToProps)(WeatherHeaderClass);
