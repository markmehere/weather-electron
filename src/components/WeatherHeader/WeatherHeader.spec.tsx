import React from 'react';
import { WeatherHeaderClass, WeatherHeaderProps } from './WeatherHeader';
import { TemperatureScale } from '../../types/TemperatureScale';
import { shallow } from 'enzyme';

describe('<WeatherHeader />', () => {

  const defaultProps: WeatherHeaderProps = {
    scale: TemperatureScale.Celsius,
    city: { value: 1105779, label: 'Sydney, AU' },
    homeCity: null,
    setUnits: jest.fn(),
    setCity: jest.fn(),
    goHome: jest.fn()
  }

  it('should render with city selector', () => {
    const result = shallow(<WeatherHeaderClass {...defaultProps}  />);
    expect(result.find('CitySelector')).toHaveLength(1);
    expect(result.find('ScaleToggler')).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const result = shallow(<WeatherHeaderClass {...defaultProps}  />);
    expect(result).toMatchSnapshot();
  });

});