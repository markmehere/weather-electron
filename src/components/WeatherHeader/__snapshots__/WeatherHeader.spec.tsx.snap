// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherHeader /> matches snapshot 1`] = `
<div
  className="header"
>
  <CitySelector
    onChange={[MockFunction]}
    value={
      Object {
        "label": "Sydney, AU",
        "value": 1105779,
      }
    }
  />
  <ScaleToggler
    onToggle={[MockFunction]}
    value={0}
  />
</div>
`;
